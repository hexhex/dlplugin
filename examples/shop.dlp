% -*- prolog -*-

plusProvides("provides","foo","moo").
plusPart("Part","cpu1").
minusPart("Part","cpu2").
minusShop("Shop","not_a_shop").

s(X) :- &dlC["shop.owl",plusPart,minusPart,plusProvides,foo,"Shop"](X).
p(X,Y) :- &dlR["shop.owl",plusPart,minusPart,plusProvides,foo,provides](X,Y).
part(X) :- &dlC["shop.owl",plusPart,minusPart,plusProvides,foo,"Part"](X).
parts1(X) :- &dlR["shop.owl",plusPart,minusPart,plusProvides,foo,"provides"]("s1",X).

nshop(X) :- &dlC["shop.owl",plusPart,minusShop,plusProvides,foo,"-Shop"](X).
