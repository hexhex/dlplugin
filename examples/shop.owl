<?xml version="1.0" encoding="ISO-8859-1"?>
<!DOCTYPE rdf:RDF []>
<rdf:RDF
  xmlns:owl="http://www.w3.org/2002/07/owl#"
  xmlns:rdf="http://www.w3.org/1999/02/22-rdf-syntax-ns#"
  xmlns:rdfs="http://www.w3.org/2000/01/rdf-schema#"
  xmlns:xsd="http://www.w3.org/2001/XMLSchema#"
  xmlns="http://www.kr.tuwien.ac.at/staff/roman/shop#"
  xml:base="http://www.kr.tuwien.ac.at/staff/roman/shop">

  <owl:Ontology rdf:ID="shop"/>

  <owl:Class rdf:ID="Shop"/>
  <owl:Class rdf:ID="Part"/>

  <owl:Class rdf:ID="Discount">
    <owl:equivalentClass>
      <owl:Restriction>
        <owl:onProperty rdf:resource="#supplier"/>
        <owl:minCardinality rdf:datatype="xsd:nonNegativeInteger">2</owl:minCardinality>
      </owl:Restriction>
    </owl:equivalentClass>
  </owl:Class>

  <owl:ObjectProperty rdf:ID="supplier">
    <rdfs:domain rdf:resource="#Shop" />
    <rdfs:range rdf:resource="#Part" />
  </owl:ObjectProperty>

  <owl:ObjectProperty rdf:ID="provides">
    <rdfs:domain rdf:resource="#Shop" />
    <rdfs:range rdf:resource="#Part" />
  </owl:ObjectProperty>

  <Shop rdf:about="#s1">
    <provides rdf:resource="#case" />
    <provides rdf:resource="#cpu" />
  </Shop>

  <Shop rdf:about="#s2">
    <provides rdf:resource="#soundcard" />
  </Shop>

  <Shop rdf:about="#s3">
    <provides rdf:resource="#memory" />
    <provides rdf:resource="#cpu" />
    <provides rdf:resource="#dvdrom" />
  </Shop>

  <Shop rdf:about="#s4">
    <provides rdf:resource="#harddisk" />
    <provides rdf:resource="#cdrom" />
  </Shop>

  <Shop rdf:about="#s5">
    <provides rdf:resource="#harddisk" />
    <provides rdf:resource="#cpu" />
  </Shop>

  <Shop rdf:about="#s6">
    <provides rdf:resource="#case" />
    <provides rdf:resource="#graphiccard" />
  </Shop>

  <Shop rdf:about="#s7">
    <provides rdf:resource="#wlan" />
    <provides rdf:resource="#harddisk" />
  </Shop>

  <Shop rdf:about="#s8">
    <provides rdf:resource="#dvdrom" />
    <provides rdf:resource="#cdrom" />
    <provides rdf:resource="#soundcard" />
    <provides rdf:resource="#memory" />
  </Shop>

  <Shop rdf:about="#s9">
    <provides rdf:resource="#case" />
    <provides rdf:resource="#harddisk" />
  </Shop>

  <Shop rdf:about="#s10">
    <provides rdf:resource="#fan" />
  </Shop>

  <Part rdf:ID="cpu" />
  <Part rdf:ID="case" />
  <Part rdf:ID="harddisk" />
  <Part rdf:ID="memory" />
  <Part rdf:ID="fan" />
  <Part rdf:ID="cdrom" />
  <Part rdf:ID="dvdrom" />
  <Part rdf:ID="soundcard" />
  <Part rdf:ID="wlan" />
  <Part rdf:ID="graphiccard" />

</rdf:RDF>
