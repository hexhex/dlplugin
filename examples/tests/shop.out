{nshop("shop:not_a_shop"), parts1("shop:cpu"), parts1("shop:case"), part("shop:cpu1"), part("shop:moo"), part("shop:fan"), part("shop:wlan"), part("shop:graphiccard"), part("shop:cdrom"), part("shop:harddisk"), part("shop:dvdrom"), part("shop:memory"), part("shop:soundcard"), part("shop:cpu"), part("shop:case"), p("shop:s2","shop:soundcard"), p("shop:s3","shop:dvdrom"), p("shop:s3","shop:memory"), p("shop:s3","shop:cpu"), p("shop:s4","shop:cdrom"), p("shop:s4","shop:harddisk"), p("shop:s8","shop:cdrom"), p("shop:s8","shop:dvdrom"), p("shop:s8","shop:memory"), p("shop:s8","shop:soundcard"), p("shop:s5","shop:harddisk"), p("shop:s5","shop:cpu"), p("shop:s6","shop:graphiccard"), p("shop:s6","shop:case"), p("shop:s7","shop:wlan"), p("shop:s7","shop:harddisk"), p("shop:s9","shop:harddisk"), p("shop:s9","shop:case"), p("shop:s10","shop:fan"), p("shop:foo","shop:moo"), p("shop:s1","shop:cpu"), p("shop:s1","shop:case"), s("shop:s2"), s("shop:s3"), s("shop:s4"), s("shop:s8"), s("shop:s5"), s("shop:s6"), s("shop:s7"), s("shop:s9"), s("shop:s10"), s("shop:foo"), s("shop:s1"), minusShop("Shop","not_a_shop"), minusPart("Part","cpu2"), plusPart("Part","cpu1"), plusProvides("provides","foo","moo")}
